modules/d/actions/vcxproj.lua

```diff
@@ -455,29 +455,9 @@
 		end
 
 		if cfg.cppdialect and cfg.cppdialect ~= "Default" then
-			local cppMap = {
-				["C++latest"] = "c++17", -- TODO: keep this up to date >_<
-				["C++98"] = "c++98",
-				["C++0x"] = "c++11",
-				["C++11"] = "c++11",
-				["C++1y"] = "c++14",
-				["C++14"] = "c++14",
-				["C++1z"] = "c++17",
-				["C++17"] = "c++17",
-				["C++2a"] = "c++20",
-				["C++20"] = "c++20",
-				["gnu++98"] = "c++98",
-				["gnu++0x"] = "c++11",
-				["gnu++11"] = "c++11",
-				["gnu++1y"] = "c++14",
-				["gnu++14"] = "c++14",
-				["gnu++1z"] = "c++17",
-				["gnu++17"] = "c++17",
-				["gnu++2a"] = "c++20",
-				["gnu++20"] = "c++20",
-			}
+			local cppMap = 	p.api.getCPPDialect()
```

Comment:
It would make more sense to just do the look up into `gcc.cxxflags.cppdialect` instead.